src/util/db/sqlstringformatter.cpp

```diff
@@ -14,7 +14,7 @@ QString SqlStringFormatter::format(
         return value; // unformatted
     }
 #if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
-    QSqlField stringField(QString(), QMetaType(QMetaType::QString));
+    QSqlField stringField((QString()), QMetaType(QMetaType::QString));
```

Comment:
Did this slip in by mistake? Please revert.

Reply:
nono actually w/o this it wouldn't compile 🤷

Reply:
what is the error you get?

Reply:
compilation error from clang that actually was telling me to do this as fix. sometimes i'm really amazing how much better are the compilation error messages nowadays compared to 10 years ago 🙃

Reply:
Please provide the exact error message or a verifiable reference that explains why this expression is needed. I have never seen these kind of extra parentheses in C++ code.

Reply:
```
/Users/wiking/mixxx/src/util/db/sqlstringformatter.cpp:17:59: error: parameter declarator cannot be qualified
    QSqlField stringField(QString(), QMetaType(QMetaType::QString));
                                               ~~~~~~~~~~~^
/Users/wiking/mixxx/src/util/db/sqlstringformatter.cpp:17:26: warning: parentheses were disambiguated as a function declaration [-Wvexing-parse]
    QSqlField stringField(QString(), QMetaType(QMetaType::QString));
                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/wiking/mixxx/src/util/db/sqlstringformatter.cpp:17:27: note: add a pair of parentheses to declare a variable
    QSqlField stringField(QString(), QMetaType(QMetaType::QString));
                          ^
                          (        )
/Users/wiking/mixxx/src/util/db/sqlstringformatter.cpp:21:16: error: member reference base type 'QSqlField (QString (*)(), QMetaType)' is not a structure or union
    stringField.setValue(value);
    ~~~~~~~~~~~^~~~~~~~~
/Users/wiking/mixxx/src/util/db/sqlstringformatter.cpp:22:33: error: reference to type 'const QSqlField' could not bind to an lvalue of type 'QSqlField (QString (*)(), QMetaType)'
    return pDriver->formatValue(stringField);
                                ^~~~~~~~~~~
```

enjoy!

Reply:
This is a decent explanation of the phenomenon: https://www.youtube.com/watch?v=ByKf_foSlXY

In this case, using uniform initialization would've worked around this I think
```suggestion
    QSqlField stringField{QString(), QMetaType(QMetaType::QString)};
```

Reply:
Even though the resolution proposed by the compiler might work this is not how I (and probably many other C++ programmers) would disambiguate the code.

Please use the solution proposed by @Swiftb0y.

Reply:
> 

LoL 🤣  will do